redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -452,6 +448,107 @@ inline bool setSeverity(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
     return true;
 }
 
+void assembleEventProperties(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const dbus::utility::DBusPropertiesMap& properties,
+                             nlohmann::json& condition,
+                             const std::string& hwStatusEventObj)
+{
+    using AssociationsValType =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+    const AssociationsValType* associations = nullptr;
+    const uint64_t* timestamp = nullptr;
+    const std::string* msgPropVal = nullptr;
+    const std::string* severity = nullptr;
+
+    const bool success = sdbusplus::unpackPropertiesNoThrow(
+        dbus_utils::UnpackErrorPrinter(), properties, "Associations",
+        associations, "Timestamp", timestamp, "Message", msgPropVal, "Severity",
+        severity);
+
+    if (!success)
+    {
+        messages::internalError(aResp->res);
+        return;
+    }
+
+    if (associations != nullptr)
+    {
+        for (const auto& assoc : *associations)
+        {
+            if (std::get<0>(assoc) == "error_log")
+            {
+                sdbusplus::message::object_path errPath = std::get<2>(assoc);
+                // we have only one condition
+                nlohmann::json::json_pointer logEntryPropPath =
+                    "/Status/Conditions/0/LogEntry"_json_pointer;
+                error_log_utils::setErrorLogUri(aResp, errPath,
+                                                logEntryPropPath, true);
+            }
+        }
+    }
+    else if (timestamp != nullptr)
+    {
+        condition["Timestamp"] = redfish::time_utils::getDateTimeStdtime(
+            static_cast<std::time_t>(*timestamp));
+    }
+    else if (msgPropVal != nullptr)
```

Comment:
Why are these else if?

Reply:
Hmm, good catch. we can only honor any one property @deepakala-k.